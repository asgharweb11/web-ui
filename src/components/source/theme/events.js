import theme from "./theme"

export const Color = (item , param) => {
    switch(item){
        case "ligth" :
            return theme.ligth.text.grey;
        case "dark" :
            return theme.dark.text.grey;
    }
}