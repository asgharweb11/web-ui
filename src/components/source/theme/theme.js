export default {
    ligth : {
        main : "white",
        content : "repeating-linear-gradient(0deg , #f2efff , #f3f3ff)",
        primary : "#2196f3",
        secondary : "#ff80ab",
        error : "#f44336",
        warning : "#ff9800",
        info : "#2196f3",
        success : "#4caf50",
        text : {
            grey : "#313131",
            grey1 : "#656565",
            grey2 : "#909090",
            grey3 : "#808080",
            grey4 : "#9a9a9a",
            grey5 : "#c7c7c7",
            color : "#ffffff",
        },
        glass : {
            grey : "#000000a3",
            grey2 : "#00000066",
            grey3 : "#0000003b",
            grey4 : "#0000001c",
            grey5 : "#0000000d",
        },
        hover : {
            primary : "#246fab",
            secondary : "#ec5286",
            error : "#dc1709",
            warning : "#e08703",
            info : "#0d7ad2",
            success : "#237b26",
        }
    },  
    dark : {
        main : "#212121",
        content : "repeating-linear-gradient(0deg ,#333333 ,#333333)",
        primary : "#4dabf5",
        secondary : "#ff99bb",
        error : "#e57373",
        warning : "#ffb74d",
        info : "#64b5f6",
        success : "#81c784",
        text : {
            grey : "#ffffff",
            grey1 : "#ececec",
            grey2 : "#c7c7c7",
            grey3 : "#212121",
            grey4 : "#6b6b6b",
            grey5 : "#464646",
            color : "2b2b2b"
        },
        glass : {
            grey : "#ffffffb3",
            grey2 : "#ffffff87",
            grey3 : "#ffffff59",
            grey4 : "#ffffff2b",
            grey5 : "#ffffff0f",
        },
        hover : {
            primary : "#90ceff",
            secondary : "#ffb5ce",
            error : "#fdaaaa",
            warning : "#ffd393",
            info : "#a8d8ff",
            success : "#b0efb2",
        }

    },
    fontSize : {
        h1 : "40px",
        xl : "25px",
        lg : "20px",
        md : "16px",
        sm : "13px",
        xs : "10px",
    },
    px : {
        p1 : "0px",
        p2 : "1px",
        p3 : "5px",
        p4 : "10px",
        p5 : "15px",
        p6 : "20px",
    },
    media : {
        mobileS: '320px',
        mobileM: '375px',
        mobileL: '425px',
        tablet: '768px',
        tabletL : "991px",
        laptop: '1024px',
        laptopM: '1200px',
        laptopL: '1440px',
        desktop: '2560px'
    }

}