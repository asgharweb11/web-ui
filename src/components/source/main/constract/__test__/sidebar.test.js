import React from "react"
// import {render , unmountComponentAtNode} from "react-dom"
// import {act} from "react-dom/test-utils";
import renderer from "react-test-renderer"
// -------------- Context
import Context from "../../../../../context"
// -------------- Styled
import { ThemeProvider } from "styled-components"
import theme from "../../../theme/theme"
// -------------- Component 
import Sidebar from "../sidebar"
// -------------- Todo
// import todo from "../../../../../context/todos"


// let container = null
// beforeEach(() => {
//     container = document.createElement("div")
//     document.body.appendChild(container)
// })

// afterEach(() => {
//     unmountComponentAtNode(container)
//     container.remove()
//     container = null
// })

test("Component : /constract/sidebar.js" , () => {
    const container = renderer.create(<Context>
        <ThemeProvider theme={theme}>
            <Sidebar />
        </ThemeProvider>
    </Context>)
        let tree = container.toJSON();
        expect(tree).toMatchSnapshot();
        // tree.children[0].children[0].props.onClick()
})