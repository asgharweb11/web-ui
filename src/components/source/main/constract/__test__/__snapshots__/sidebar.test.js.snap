// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component : /constract/sidebar.js 1`] = `
<div
  className="sc-bdfBwQ rVKDb"
>
  <ul>
    <li
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-search fa-w-16 fa-lg "
        data-icon="search"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </li>
  </ul>
  <ul>
    <li>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-utensils fa-w-13 fa-lg "
        data-icon="utensils"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 416 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M207.9 15.2c.8 4.7 16.1 94.5 16.1 128.8 0 52.3-27.8 89.6-68.9 104.6L168 486.7c.7 13.7-10.2 25.3-24 25.3H80c-13.7 0-24.7-11.5-24-25.3l12.9-238.1C27.7 233.6 0 196.2 0 144 0 109.6 15.3 19.9 16.1 15.2 19.3-5.1 61.4-5.4 64 16.3v141.2c1.3 3.4 15.1 3.2 16 0 1.4-25.3 7.9-139.2 8-141.8 3.3-20.8 44.7-20.8 47.9 0 .2 2.7 6.6 116.5 8 141.8.9 3.2 14.8 3.4 16 0V16.3c2.6-21.6 44.8-21.4 48-1.1zm119.2 285.7l-15 185.1c-1.2 14 9.9 26 23.9 26h56c13.3 0 24-10.7 24-24V24c0-13.2-10.7-24-24-24-82.5 0-221.4 178.5-64.9 300.9z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </li>
    <li>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-map fa-w-18 fa-lg "
        data-icon="map"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 117.66v346.32c0 11.32 11.43 19.06 21.94 14.86L160 416V32L20.12 87.95A32.006 32.006 0 0 0 0 117.66zM192 416l192 64V96L192 32v384zM554.06 33.16L416 96v384l139.88-55.95A31.996 31.996 0 0 0 576 394.34V48.02c0-11.32-11.43-19.06-21.94-14.86z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </li>
    <li>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chart-pie fa-w-17 fa-lg "
        data-icon="chart-pie"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 544 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M527.79 288H290.5l158.03 158.03c6.04 6.04 15.98 6.53 22.19.68 38.7-36.46 65.32-85.61 73.13-140.86 1.34-9.46-6.51-17.85-16.06-17.85zm-15.83-64.8C503.72 103.74 408.26 8.28 288.8.04 279.68-.59 272 7.1 272 16.24V240h223.77c9.14 0 16.82-7.68 16.19-16.8zM224 288V50.71c0-9.55-8.39-17.4-17.84-16.06C86.99 51.49-4.1 155.6.14 280.37 4.5 408.51 114.83 513.59 243.03 511.98c50.4-.63 96.97-16.87 135.26-44.03 7.9-5.6 8.42-17.23 1.57-24.08L224 288z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </li>
    <li>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-w-14 fa-lg "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </li>
  </ul>
  <ul>
    <li>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-shopping-cart fa-w-18 fa-lg "
        data-icon="shopping-cart"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </li>
    <li>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-gem fa-w-18 fa-lg "
        data-icon="gem"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M485.5 0L576 160H474.9L405.7 0h79.8zm-128 0l69.2 160H149.3L218.5 0h139zm-267 0h79.8l-69.2 160H0L90.5 0zM0 192h100.7l123 251.7c1.5 3.1-2.7 5.9-5 3.3L0 192zm148.2 0h279.6l-137 318.2c-1 2.4-4.5 2.4-5.5 0L148.2 192zm204.1 251.7l123-251.7H576L357.3 446.9c-2.3 2.7-6.5-.1-5-3.2z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </li>
    <li>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-cog fa-w-16 fa-lg "
        data-icon="cog"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </li>
  </ul>
</div>
`;
